sled-agent/src/bootstrap/early_networking.rs

```diff
@@ -38,59 +44,185 @@ pub enum EarlyNetworkSetupError {
     Dendrite(String),
 
     #[error("Error during DNS lookup: {0}")]
-    DnsResolver(#[from] internal_dns::resolver::ResolveError),
+    DnsResolver(#[from] ResolveError),
 }
 
 /// Code for configuring the necessary network bits to bring up the control
 /// plane
-pub struct EarlyNetworkSetup {
-    log: Logger,
+pub struct EarlyNetworkSetup<'a> {
```

Comment:
Seems a touch overkill to parameterize on a lifetime just for a logger.

Reply:
Eh, maybe, but nobody actually stores one of these things; it's really just a way to namespace some functions. I also considered just removing `EarlyNetworkSetup` altogether, and making its methods free functions that take `&Logger`s. Would you prefer that?

Reply:
Either way is fine with me. I made it a type originally because I thought it was going to get stored in bootstrap agent as a minimum. Feel free to leave it as is. I should have marked it `nit`